repokid/datasource/plugin.py

```diff
@@ -0,0 +1,67 @@
+#  Copyright 2021 Netflix, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import Dict
+from typing import Generic
+from typing import ItemsView
+from typing import Iterable
+from typing import Iterator
+from typing import KeysView
+from typing import Optional
+from typing import TypeVar
+from typing import ValuesView
+from typing import cast
+
+from repokid.plugin import RepokidPlugin
+from repokid.types import RepokidConfig
+
+logger = logging.getLogger("repokid")
+KT = TypeVar("KT")
```

Comment:
nit: Should KT and VT be generics, and put in a different file, so they can be re-used in the future?

Reply:
Moved them to `repokid.types`. They're supposed to be `TypeVar`, and then `Generic` is used in the class definition, i.e.
```python
class DatasourcePlugin(RepokidPlugin, Generic[KT, VT]):
    [...]
```